### DISCUSSION TITLE
Deploys are passing even if "npm install" fails

### DISCUSSION THREAD

**Discussion by jgtvares** [vercel/community#1053] [Help] [answered]
# Problem
I've been having this problem a lot.
When dependencies installation fails the build continues and the project is successfully deployed.
Then my app crashes with a lot of `MODULE_NOT_FOUND` errors.
This was not happening before (2 or 3 months ago).

# More details
To get more into details, the error I often get with `npm install` is a **401 Unable to authenticate**.
That happens because my project uses a couple of private libraries that I developed. 
In order to install them successfully, I'm using the `NPM_RC` environment variable to set npm token for authentication in my private repository.
This token expires after a while and I need to update the variable from time to time, that's why `npm install` fails sometimes.
Before, deploys used to fail because of this, then I knew I had to update the npm token.
But now, for some reason, this error gets bypassed and my project crashes and that's a huge problem!

Is anyone also going through this?

<img width="724" alt="image" src="https://user-images.githubusercontent.com/47645535/201676038-2c919358-c399-4916-9ac9-a7896250dd99.png">

**Comment by amyegan** [comment, marked as answer]
Hi @jgtvares! Are you still seeing this behavior? If so, I would like to take a closer look at the problem. Can you share more details about your project, a deployment ID (from the deployment details URL), or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example)?

**Reply by jgtvares** [reply]
Hey @amyegan!
Yes I'm still experiencing this.

Here's the ID of one of the broken deployments: **GzRBi7FUGy6rh9hzZ5CZWWYmqiX8**

**Reply by amyegan** [reply]
Thanks for sharing this! It looks like you also opened an issue (https://github.com/vercel/vercel/issues/8901) about this and I just saw that Swarnava added an update there. The engineers have implemented a bugfix (https://github.com/vercel/vercel/pull/8909) that will solve this in the next CLI release 🚀
